runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/middleware/Retry.kt

```diff
@@ -31,29 +32,32 @@ public open class Retry<O>(
     override suspend fun <H : Handler<SdkHttpRequest, O>> handle(request: SdkHttpRequest, next: H): O =
         if (request.subject.isRetryable) {
             var attempt = 1
-            val logger = request.context.getLogger("Retry")
-            val wrappedPolicy = PolicyLogger(policy, logger)
+            val wrappedPolicy = PolicyLogger(policy, request.context)
             val outcome = strategy.retry(wrappedPolicy) {
-                if (attempt > 1) {
-                    logger.debug { "retrying request, attempt $attempt" }
-                }
+                request.context.childTraceSpan("Attempt-$attempt") {
+                    if (attempt > 1) {
+                        request.context.getLogger<Retry<*>>().debug { "retrying request, attempt $attempt" }
+                    }
 
-                // Deep copy the request because later middlewares (e.g., signing) mutate it
-                val requestCopy = request.deepCopy()
+                    // Deep copy the request because later middlewares (e.g., signing) mutate it
+                    val requestCopy = request.deepCopy()
 
-                onAttempt(requestCopy, attempt)
-                when (val body = requestCopy.subject.body) {
-                    // Reset streaming bodies back to beginning
-                    is HttpBody.Streaming -> body.reset()
-                    else -> {}
-                }
+                    onAttempt(requestCopy, attempt)
+                    when (val body = requestCopy.subject.body) {
+                        // Reset streaming bodies back to beginning
+                        is HttpBody.Streaming -> body.reset()
+                        else -> {}
+                    }
 
-                attempt++
-                next.call(requestCopy)
+                    attempt++
+                    next.call(requestCopy)
+                }
             }
             outcome.getOrThrow()
         } else {
-            next.call(request)
+            request.context.childTraceSpan("Attempt-1") { // Include an attempt trace span even though we won't retry
```

Comment:
Perhaps a different span name that is more meaningful like `NonRetryableAttempt`?